### PR TITLE
Default conda environments: Install CloudPickle from pip

### PR DESCRIPTION
**Pull request by dbczumar**
We've had several test failures and user errors occur because updates to Conda packages tend to lag behind updates to PyPI package. As a result, when a user / test suite installs the latest version of a package from PyPI, our default conda environments for models often reference dependency versions that do not yet exist on Conda.

~~This PR modifies the default conda environments to install most major dependencies from PyPI instead of Conda, hopefully ameliorating this issue.~~

As a small step toward ameliorating this issue, this PR modifies default conda environments to install the `CloudPickle` library from PyPI instead of Conda, since updates to this library often hit PyPI before Conda.

### CODE REVIEW

File: mlflow/keras.py [hunk 1]
```diff
@@ -27,14 +27,14 @@
 FLAVOR_NAME = "keras"
 
 DEFAULT_CONDA_ENV = _mlflow_conda_env(
-    additional_conda_deps=[
```
Comment (tomasatdatabricks) [inline mlflow/keras.py]: There is a difference in tensorflow from pip / conda.  I think the conda tensorflow is supposed be built with mkl support and is support to be better on a cpu.

**Review by tomasatdatabricks — COMMENTED** [review]
Looks good but I am not sure whether we want to this for tensorflow. I believe there is a difference in tensorflow for cpu from conda vs pip.

**Comment by acroz** [comment]
There are definitely advantages to installing from conda as @tomasatdatabricks says, not just for Tensorflow/Keras but also for things like scikit-learn which will benefit from the MKL-linked numpy that conda provides.

The alternative to this solution would be to install those test dependencies which are used in this way from conda when running tests in CI. That way, the versions written in default conda environments will definitely exist on conda.

**Review by tomasatdatabricks — COMMENTED** [review]
Looks like one of the tests needs to be updated. Otherwise lgtm.

**Review by tomasatdatabricks — APPROVED** [review]
lgtm!

**Pull request merged**